src/Common/src/Interop/Unix/Interop.IOErrors.cs

```diff
@@ -77,41 +78,41 @@ internal static Exception GetExceptionForIoErrno(int errno, string path = null,
                         new FileNotFoundException(SR.IO_FileNotFound);
                 }
 
-            case Errors.EACCES:
-            case Errors.EBADF:
-            case Errors.EPERM:
+            case Error.EACCES:
+            case Error.EBADF:
+            case Error.EPERM:
                 return !string.IsNullOrEmpty(path) ?
                     new UnauthorizedAccessException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, path)) :
                     new UnauthorizedAccessException(SR.UnauthorizedAccess_IODenied_NoPathName);
 
-            case Errors.ENAMETOOLONG:
+            case Error.ENAMETOOLONG:
                 return new PathTooLongException(SR.IO_PathTooLong);
 
-            case Errors.EWOULDBLOCK:
+            case Error.EWOULDBLOCK:
                 return !string.IsNullOrEmpty(path) ?
-                    new IOException(SR.Format(SR.IO_SharingViolation_File, path), errno) :
-                    new IOException(SR.IO_SharingViolation_NoFileName, errno);
+                    new IOException(SR.Format(SR.IO_SharingViolation_File, path), errorInfo.Errno) :
+                    new IOException(SR.IO_SharingViolation_NoFileName, errorInfo.Errno);
 
-            case Errors.ECANCELED:
+            case Error.ECANCELED:
                 return new OperationCanceledException();
 
-            case Errors.EFBIG:
+            case Error.EFBIG:
                 return new ArgumentOutOfRangeException("value", SR.ArgumentOutOfRange_FileLengthTooBig);
 
-            case Errors.EEXIST:
+            case Error.EEXIST:
                 if (!string.IsNullOrEmpty(path))
                 {
-                    return new IOException(SR.Format(SR.IO_FileExists_Name, path), errno);
+                    return new IOException(SR.Format(SR.IO_FileExists_Name, path), errorInfo.Errno);
                 }
                 goto default;
 
             default:
-                return GetIOException(errno);
+                return GetIOException(errorInfo);
         }
     }
 
-    internal static Exception GetIOException(int errno)
+    internal static Exception GetIOException(Interop.ErrorInfo errorInfo)
     {
-        return new IOException(libc.strerror(errno), errno);
+        return new IOException(errorInfo.GetErrorMessage(), errorInfo.Errno);
```

Comment:
I went back and forth on whether it should be erroInfo.Errno or errorInfo.Error, but I think you're right that it should be Errno.  Even though it's platform-specific, the platform's errno value will make more sense to a developer than what are effectively internal values used in Error.